### ISSUE TITLE
App crashes, does not restart

### ISSUE THREAD

**Issue by Barbados02** [rogierhofboer/com.ikea.tradfri.gateway#20]
The app crashes daily, a manual restart of the app makes it work again.
Before i did this with candy, but this does not work since homey v2.

This is the error i see in the homey app.

![image](https://user-images.githubusercontent.com/25595391/51590899-d919b800-1eeb-11e9-8c8d-dba5c022a1fd.png)

**Comment by rogierhofboer** [comment]
@Barbados02 thanks for the report. It seems to crash in in the underlying node-tradfri-client module.
Since I am not on 2.0 myself I cannot reproduce the issue.
Just to be sure: Did you change scenes / groups in between or is it running without making changes and then it suddenly crashes?

**Comment by Barbados02** [comment]
I waited until it happend again. It seems the app crashed  (just?) after i turned on the light.  
unfortunately Homey is not very exact in the time of crash that i find.
In the log what Homey does see screenshot:

![image](https://user-images.githubusercontent.com/25595391/51679167-6fc6a180-1fde-11e9-8e22-2c664595e448.png)

And the app time of the crash is not exact at all but:

![image](https://user-images.githubusercontent.com/25595391/51679212-9553ab00-1fde-11e9-9e45-4c44001e122b.png)

**Comment by rogierhofboer** [comment]
Could be, but most probably it has a bad ZigBee reception due to the distance of it to the gateway or other bulbs. Still the app should not crash. 

As soon as 2.x also solves the z-wave stability / range problems, I will upgrade.

**Comment by Barbados02** [comment]
Strange thing is that the tratfri gateway works without a problem. 
I only send on or off to a room i did not add all bulbs seperate.
So i do not think it has to do with zigbee because homey is not talking zigbee to the gatweway, or do i miss something?

**Comment by rogierhofboer** [comment]
You are correct about that. Homey communicates over TCP (via Wifi / ethernet) to the tradfri gateway and the tradfri gateway communicaties with Zigbee to the bulbs.
I am now on Homey firmware 2.04 myself

**Comment by Lindansaren** [comment]
It happens to me each day as well. Just random crashes of the app. All works fine after rebooting the app in homey.

**Comment by cesarenl** [comment]
Any updates on the functionality of this app? I'm postponing the update to Homey 2, till I'm pretty sure most of the apps I'm using are working.

**Comment by rogierhofboer** [comment]
I still did not take take the time to look at it. 

The problem is, I don't really use the gateway myself now. 

Hue colour lamps stopped working correctly with the Ikea gateway after the last update and also when switched via regular switch, the lamps 'flicker' once or twice a few seconds after they are turned on when using the Ikea gateway.

I might try to update the node-tradfri-client that I am using, because this app is only a wrapper around this lib.

**Comment by cesarenl** [comment]
Is there another way to control the tradfri's via Homey, without the gateway?

**Comment by rogierhofboer** [comment]
There are 4 ways to control them:

1. Directly via Homey
2. Via the Ikea Tradfri gateway
3. Via the Philips Hue Bridge
4. Not using Homey at all (i.g. raspberry pi with a Conbee)

Note that the last time I tried (more than a year ago) all the options, changing themes (colour / light temperature) with 5 button remote only worked with options 2 and 4, in other works, if you use the Hue bridge or Homey directy you loose some functionality of the 5 button remote.

**Comment by Berrie1** [comment]
I am experiencing the same. When the light switch is used to turn on the lights, the app crashes. When I restart the app, everything works fine again. The problem occurs since 2 more light bulbs were added. The total number of bulbs was increased from 10 to 12. The two added bulbs are a newer version than the other bulbs.
Would it help to send a diagnostic report?

**Comment by json81** [comment]
Anyone got problems with crashes anymore? It works fine for me.

**Comment by Berrie1** [comment]
After the group of 12 bulbs was split in 2 groups of 6 bulbs, I am not experiencing any problems anymore. So it now works fine for me as well.

**Comment by rogierhofboer** [comment]
I still get many reports of crashes in the (external) node-tradfri-client. The stack trace is always the one at the start of this issue.
Because I don't experience the problem, maybe because I don't use groups myself, please contact me (via this issue or via email) so we can look at it.

**Comment by rogierhofboer** [comment]
Underlying issue:
https://github.com/AlCalzone/node-tradfri-client/issues/36

**Comment by rogierhofboer** [comment]
Maybe 'fixed' by the new scenes ;)
